src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -417,6 +417,97 @@ public ImmutableArray<T> AddRange(IEnumerable<T> items)
             return self.InsertRange(self.Length, items);
         }
 
+        /// <summary>
+        /// Adds the specified items to the end of the array.
+        /// </summary>
+        /// <param name="items">The items.</param>
+        /// <param name="length">The number of elements from the source array to add.</param>
+        /// <returns>The new immutable collection.</returns>
+        public ImmutableArray<T> AddRange(T[] items, int length)
+        {
+            var self = this;
```

Comment:
Can you explain why you explicitly assign `this` to the variable `self`? Do we need `self`?

Reply:
I followed the code style from other methods in this file to make it consistent.